pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -407,6 +407,9 @@ def training_type_plugin(self) -> TrainingTypePlugin:
         # attach checkpoint plugin to the training type plugin
         if self._checkpoint_io is not None:
             self._training_type_plugin.checkpoint_io = self._checkpoint_io
+        precision_plugin = self.precision_plugin
+        if precision_plugin is not None:
+            self._training_type_plugin._precision_plugin = self.precision_plugin
```

Comment:
```suggestion
        if self.precision_plugin is not None:
            self._training_type_plugin._precision_plugin = self.precision_plugin
```

Reply:
There is selection logic build in property precision plugin(): It's better call it once instead multiple times

https://github.com/PyTorchLightning/pytorch-lightning/blob/c5cabf836e0729bd48c16f538d48bb72651a73b8/pytorch_lightning/trainer/connectors/accelerator_connector.py#L393-L397

The self.select_precision_plugin() is quite heavy logic

Reply:
Sounds good @four4fish , just a note, it is currently also used multiple times, so you could make it also 

```suggestion
        precision_plugin = self.precision_plugin
        if precision_plugin is not None:
            self._training_type_plugin._precision_plugin = precision_plugin
```

Reply:
Yeah! typo!! thank you for catching it :)